### PR TITLE
feat: in-app embed settings

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description

Adds a code snippet for embedding with iframes.

Fixes https://github.com/appsmithorg/appsmith/issues/18292

Media

https://user-images.githubusercontent.com/67054171/209044270-d3b2d03d-3283-4edc-8561-d5381dfac4ff.mov

#### Code changes
- Most of the changes in un related cypress files are formatting changes made by the pre-commit hook  For e.g https://github.com/appsmithorg/appsmith/pull/19023/files#diff-8512d8d522be6cbd62765ee75c2ec23fd1c103de0834808b60f0108b1f36a6daR8
- Chrome asks for permission while copying to clipboard on cypress added a command to grant the same https://github.com/appsmithorg/appsmith/pull/19023/files#diff-3eda263cbeef4ddbace0cb7244ba4ccbf097968e297f271bf403a60967e75b1dR33
- Will get a new release from design system before merging https://github.com/appsmithorg/appsmith/pull/19023/files#diff-b8d2936a034f3ef827722f0cd18b81d5522b816bfe559ce499e8841f2142aeafR49
- Updating margins of tab when there is no message. For group selection on ee we pass a message prop keeping the margins  as it is in that case. https://github.com/appsmithorg/appsmith/pull/19023/files#diff-2a32a9d6d34facd2317a42d447073d12dd2a944903979bf6bd7824c230a34570R30
- Created a hook to reuse logic as we have two slight variations of the same UI https://github.com/appsmithorg/appsmith/pull/19023/files#diff-7a5ea9c499be5787a047fe5fb52c1e97c747689b5c0af83de09e39c5730922f1R39
- Moved switch styles to components as we use the same styles in page settings and embed settings https://github.com/appsmithorg/appsmith/pull/19023/files#diff-614654ec58d14187e37f4c41ceb094d4a660a191a6b8b9cea77810b1737c633aR4
- Removed title etc for the invite modal opened from edit mode header https://github.com/appsmithorg/appsmith/pull/19023/files#diff-88af4adc81bc30e3f31926240eb347829dc88ba7bd516ea08585b44e1c1d9ba4R475 
- Changes to test file in embed settings to get the deploy url from clipboard, previously it was picking from an input https://github.com/appsmithorg/appsmith/pull/19023/files#diff-f1b14022438d58b8616b2fbff67a259726a5b50e1952f458e449e198ee351ec7R33
- Removed duplicate scroll which appears outside the invite modal(visible in release and prod)https://github.com/appsmithorg/appsmith/pull/19023/files#diff-8c6a7a48042ef3b85d76eaad387ec54cd18df8eef5c99985e29f66bef8694ed8L578
- Updated to latest design system version as there is a dependency on it. 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual
 Self hosted dp details is on https://github.com/appsmithorg/appsmith/issues/18292#issue-1457319773
 Cloud dp link can taken from the PR
- Cypress

### Issues raised during DP testing
Issues are raised are from https://github.com/appsmithorg/appsmith/issues/18292#issuecomment-1357089835 comment on wards

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [x] Test plan has been approved by relevant developers
- [x] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/AppSettingsPane/AppSettings/index.tsx [hunk 1]
```diff
@@ -73,21 +81,31 @@ function AppSettings() {
       id: "t--general-settings-header",
       icon: "settings-2-line",
       isSelected: selectedTab.type === AppSettingsTabs.General,
-      name: GENERAL_SETTINGS_SECTION_HEADER(),
+      name: createMessage(GENERAL_SETTINGS_SECTION_HEADER),
       onClick: () => {
         setSelectedTab({ type: AppSettingsTabs.General });
       },
-      subText: GENERAL_SETTINGS_SECTION_HEADER_DESC(),
+      subText: createMessage(GENERAL_SETTINGS_SECTION_HEADER_DESC),
```
Comment (eco-monk) [inline app/client/src/pages/Editor/AppSettingsPane/AppSettings/index.tsx:88]: Just noticing this. General sub text should't contain share.
image

@RoopKrrish9696 can help re-frame this.

**Review by eco-monk — COMMENTED** [review]
Mostly nitpicks. I would suggest getting cypress cases reviewed by someone from QA automation.

**Review by ankitakinger — APPROVED** [review]
Invite user modal changes look good to me. Please get it tested though.

File: app/client/src/pages/Editor/AppSettingsPane/AppSettings/index.tsx (hunk 1 above)
Comment (akash-codemonk) [reply app/client/src/pages/Editor/AppSettingsPane/AppSettings/index.tsx:88]: Moving with this for now. We can take this up separately.

**Review by Aishwarya-U-R — COMMENTED** [review]
More coverage needed for EmbedSettings. Cypress changes looks ok otherwise.

**Comment by akash-codemonk** [comment]
> https://github.com/appsmithorg/appsmith/actions/runs/3769857348

Added more tests

**Comment by Aishwarya-U-R** [comment]
@akash-codemonk - Show navigation bar is not working in sync between the AppSettings pane & Share modal.. Is this expected?

https://user-images.githubusercontent.com/91450662/209430097-dc7df368-a849-4a8a-afa3-0d9f13263fe0.mov

**Comment by ankitakinger** [comment]
@akash-codemonk have we also tested this share modal in EE codebase? If not, I highly recommend you to do so. Please get it tested on EE codebase along with your changes.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/SettingsPane/EmbedSettings_spec.ts [hunk 2]
```diff
@@ -19,11 +19,22 @@ describe("In-app embed settings", () => {
...
+
+    _.inviteModal.OpenShareModal();
+    _.inviteModal.SelectEmbedTab();
+
+    _.agHelper
+      .GetElement(_.embedSettings.locators._showNavigationBar)
+      .should("be.checked");
+
+    _.embedSettings.ToggleShowNavigationBar("false");
+    _.inviteModal.CloseModal();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/SettingsPane/EmbedSettings_spec.ts:38]: Please also check for untoggling between App Pane & Embed in Share modal.. that was were the issue was..

**Comment by akash-codemonk** [comment]
> @akash-codemonk have we also tested this share modal in EE codebase? If not, I highly recommend you to do so. Please get it tested on EE codebase along with your changes. Also, make sure this is tested on home page + RBAC screens.

@ankitakinger Yes have checked it before and have mentioned in the PR description. The only place where it could affect are margins for the tabs in the modal. I am keeping it as it is for ee as it has a message(i.e it has other content above tabs) https://github.com/appsmithorg/appsmith/pull/19023/files#diff-2a32a9d6d34facd2317a42d447073d12dd2a944903979bf6bd7824c230a34570R30. Tabs are used in ee for adding user groups/roles. I have rechecked again now.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/SettingsPane/EmbedSettings_spec.ts (hunk 2 above)
Comment (akash-codemonk) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/SettingsPane/EmbedSettings_spec.ts]: > @akash-codemonk - Show navigation bar is not working in sync between the AppSettings pane & Share modal.. Is this expected?

Actually it does sync but there is a small delay. Have fixed it.


> Please also check for untoggling between App Pane & Embed in Share modal.. that was were the issue was..

Added tests for both.

**Comment by Aishwarya-U-R** [comment]
@akash-codemonk - Toggling off ShowNav from App pane is not actually hiding the Navigation bar in Deployed mode.. is this expected? But it syns value with Share modal.

https://user-images.githubusercontent.com/91450662/209446128-64412a3c-763a-4d6a-9088-7a666090ed7c.mov

**Comment by akash-codemonk** [comment]
> @akash-codemonk - Toggling off ShowNav from App pane is not actually hiding the Navigation bar in Deployed mode.. is this expected? But it syns value with Share modal.
> 
>  Screen.Recording.2022-12-24.at.11.01.24.PM.mov

It won't change the view of the deployed version. This is only for previewing the final result of the generated snippet. To preview you need to click on the preview embed link in modal.

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good!

**Comment by akash-codemonk** [comment]
@ankitakinger @eco-monk @Aishwarya-U-R Gentle ping for approval if all looks good, so that I can merge it before freeze.

**Review by ankitakinger — APPROVED** [review]
LGTM. Please wait for Aishwarya and Anand's approval as well though.

**Comment by eco-monk** [comment]
LGTM.! 
I'm on mobile couldn't find an option to approve.

**Comment by akash-codemonk** [comment]
> LGTM. Please wait for Aishwarya and Anand's approval as well though.

@Aishwarya-U-R has reviewed already. https://github.com/appsmithorg/appsmith/pull/19023#pullrequestreview-1229683059

**Pull request merged**